cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -1364,15 +1364,41 @@ impl fmt::Debug for Inst {
     }
 }
 
+/// Is the given (AluRmiROpcode, src, dst) combination causing a redefinition of the input operand,
+/// instead of an actual use?
+fn alu_rmi_r_is_def(src: &RegMemImm, dst: Writable<Reg>, op: AluRmiROpcode) -> bool {
+    op == AluRmiROpcode::Xor
+        && if let RegMemImm::Reg { reg: src } = src {
+            *src == dst.to_reg()
+        } else {
+            false
+        }
+}
+
+/// Is the given (SseOpcode, src, dst) combination causing a redefinition of the input operand,
+/// instead of an actual use?
+fn xmm_rm_r_is_def(src: &RegMem, dst: Writable<Reg>, op: SseOpcode) -> bool {
+    (op == SseOpcode::Xorps || op == SseOpcode::Xorpd)
```

Comment:
Also `cmpeqxx` or whatever the recommended generate-all-1s idiom is.  The Intel opt guide suggests this is recommended for Sandy Bridge onwards.

Reply:
I add that in #2060.